sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -54,6 +55,8 @@ class QueryExecution(
     val logical: LogicalPlan,
     val tracker: QueryPlanningTracker = new QueryPlanningTracker) extends Logging {
 
+  val id: Long = QueryExecution.nextExecutionId
```

Comment:
What is the use case to add this ID?

Reply:
We have a listener which parses the logical & physical plan from the query execution and reflects the changes to Apache Atlas. While it works well in general, there's a pain point on the listener due to the fact Spark can callback the listener multiple times on the same query execution, whereas the instance of query execution doesn't have the unique ID to distinct. That's already I wrote in PR description as well as JIRA description.